### PR TITLE
Faster native NTT

### PR DESCRIPTION
**Pull request by fboemer**
Speeds up native NTT by:
* For MSVC, manually unrolling loops doesn't support `#pragma loop unroll` or similar
* Extracting separate `t=1, 2, 4, 8` loops

With these changes, I'm seeing on a Mac:
Before:
Screen Shot 2021-08-13 at 12 10 06 PM

After:
Screen Shot 2021-08-13 at 12 08 40 PM

Also standardize `W_op` to `W` and `(W_precon | W_op_precon)` to `W_precon` and fix a few documentation bugs

### CODE REVIEW

File: hexl/ntt/fwd-ntt-avx512.cpp [hunk 1]
```diff
@@ -214,28 +215,6 @@ void FwdT8(uint64_t* operand, __m512i v_neg_modulus, __m512i v_twice_mod,
   }
 }
 
-/// @brief AVX512 implementation of the forward NTT
```
Comment (fboemer) [inline hexl/ntt/fwd-ntt-avx512.cpp:217]: Documentation already present in header, no need to duplicate here.

File: hexl/ntt/inv-ntt-avx512.cpp [hunk 2]
```diff
@@ -215,28 +215,6 @@ void InvT8(uint64_t* operand, __m512i v_neg_modulus, __m512i v_twice_mod,
   }
 }
 
-/// @brief AVX512 implementation of the inverse NTT
```
Comment (fboemer) [inline hexl/ntt/inv-ntt-avx512.cpp:218]: Documentation already in header, no need t  duplicate here

File: hexl/ntt/inv-ntt-avx512.hpp [hunk 3]
```diff
@@ -14,15 +14,15 @@ namespace hexl {
 /// @param[in, out] operand Input data. Overwritten with NTT output
 /// @param[in] n Size of the transfrom, i.e. the polynomial degree. Must be a
 /// power of two.
-/// @param[in] modulus Prime modulus. Must satisfy q == 1 mod 2n
-/// @param[in] root_of_unity_powers Powers of inverse 2n'th root of unity in
+/// @param[in] modulus Prime modulus q. Must satisfy q == 1 mod 2n
+/// @param[in] inv_root_of_unity_powers Powers of inverse 2n'th root of unity in
```
Comment (fboemer) [inline hexl/ntt/inv-ntt-avx512.hpp:18]: Doxygen would usually catch this, except we only require docstrings for public header files, i.e. those in `hexl/include`.

**Review by GelilaSeifu — APPROVED** [review]
LGTM

File: hexl/ntt/fwd-ntt-avx512.hpp [hunk 4]
```diff
@@ -14,15 +14,15 @@ namespace hexl {
 /// @param[in, out] operand Input data. Overwritten with NTT output
 /// @param[in] n Size of the transfrom, i.e. the polynomial degree. Must be a
```
Comment (jlhcrawford) [inline hexl/ntt/fwd-ntt-avx512.hpp:15]:
```suggestion
/// @param[in] n Size of the transform, i.e. the polynomial degree. Must be a
```

File: hexl/ntt/inv-ntt-avx512.hpp [hunk 5]
```diff
@@ -14,15 +14,15 @@ namespace hexl {
 /// @param[in, out] operand Input data. Overwritten with NTT output
 /// @param[in] n Size of the transfrom, i.e. the polynomial degree. Must be a
```
Comment (jlhcrawford) [inline hexl/ntt/inv-ntt-avx512.hpp:15]:
```suggestion
/// @param[in] n Size of the transform, i.e. the polynomial degree. Must be a
```

**Review by jlhcrawford — CHANGES_REQUESTED** [review]
Some typos and comments. 

What is the convention we are choosing regarding the copyright of new files? I notice that `ntt-default.{h,c}pp` are new files yet have a copyright from 2020-2021. Are we creating copyrights for the years at which the files exist (i.e. 2021 only for the files in question) or keeping them the same across all files regardless of the lifetime of the file itself?

There are a couple other typos I've noticed just outside the "comment-able" areas. I'll fix them myself to avoid the hassle.

**Review by jlhcrawford — APPROVED** [review]
LGTM

**Pull request merged**